src/parser/Lexer.cpp

```diff
@@ -2108,34 +2219,38 @@ static ALWAYS_INLINE KeywordKind isKeyword(const StringBufferAccessData& data)
 ALWAYS_INLINE void Scanner::scanIdentifier(Scanner::ScannerResult* token, char16_t ch0)
 {
     ASSERT(token != nullptr);
-    Token type;
+    Token type = Token::IdentifierToken;
     const size_t start = this->index;
 
     // Backslash (U+005C) starts an escaped character.
     ScanIDResult id = UNLIKELY(ch0 == 0x5C) ? this->getComplexIdentifier() : this->getIdentifier();
+    const auto& data = std::get<0>(id);
     const size_t end = this->index;
 
     // There is no keyword or literal with only one character.
     // Thus, it must be an identifier.
-    KeywordKind keywordKind;
-    const auto& data = std::get<0>(id);
-    if (data.length == 1) {
-        type = Token::IdentifierToken;
-    } else if ((keywordKind = isKeyword(data))) {
-        token->setKeywordResult(this->lineNumber, this->lineStart, start, this->index, keywordKind);
-        return;
-    } else if (data.length == 4) {
-        if (data.equalsSameLength("null")) {
+    if (data.length > 1) {
+        KeywordKind keywordKind = getKeyword(data);
+
+        token->secondaryKeywordKind = keywordKind;
+
+        switch (keywordKind) {
+        case NotKeyword:
+            break;
+        case NullKeyword:
             type = Token::NullLiteralToken;
-        } else if (data.equalsSameLength("true")) {
-            type = Token::BooleanLiteralToken;
-        } else {
-            type = Token::IdentifierToken;
+            break;
+        case TrueKeyword:
+        case FalseKeyword:
+            type = BooleanLiteralToken;
+            break;
+        default:
+            if (keywordKind >= StrictModeReservedWord && keywordKind != YieldKeyword && keywordKind != LetKeyword) {
+                break;
+            }
+            token->setKeywordResult(this->lineNumber, this->lineStart, start, this->index, keywordKind);
```

Comment:
Oh, I got it. You use fall through for the default case to avoid duplicated `setKeywordResult` statement.
Then, what about following codes?

```c++
case YieldKeyword:
case LetKeyword:
  token->setKeywordResult(this->lineNumber, this->lineStart, start, this->index, keywordKind);
default:
  if (keywordKind >= StrictModeReservedWord) {
    break;
  }
  token->setKeywordResult(this->lineNumber, this->lineStart, start, this->index, keywordKind);
```

Reply:
If the compiler is smart, it will merge the code paths. My code was for the less smart compilers :) I will change it according to your suggestion.